### ISSUE TITLE
[BUG] [Linter] H037 false positive on attribute names containing a dot

### ISSUE THREAD

**Issue by silverwind** [djlint/djLint#2352]
- [x] I'm on the [latest version](https://pypi.org/project/djlint/) of djLint
- [x] I've searched the [issues](https://github.com/djlint/djLint/issues)
- [x] I've read the [docs](https://djlint.com)

## System Info

- OS: macOS 15.7.3
- Python Version: 3.13.1
- djLint Version: 1.43.1
- template language: html

## Issue

H037 reports a duplicate when two attribute names differ only before a `.`. HTML permits `.` in attribute names, so these are two distinct attributes.

The name pattern in `djlint/rules/H037.py` is `[-:a-z_][-:\w]*`, which excludes `.`. The match therefore starts after the dot and both names register as `checked`. Adding `.` to that character class fixes it.

`<a data-a.checked="1" checked>x</a>` is not reported, since a bare attribute has no `=` and is never collected, so the mangled names are only ever compared against each other.

## How To Reproduce

`repro.html`:

```html
<a data-a.checked="1" data-b.checked="2">x</a>
```

```
$ djlint --profile html --lint repro.html
H037 1:10 Duplicate attribute found. checked
```

Expected: no error.

## Contents of djlint.toml/.djlintrc/pyproject.toml [tool.djlint]

None, reproduced with `--profile html` alone.

_Filed with Claude Code._
